File: meridian-web/frontend/src/api/contextPreview.ts
import api from './client';

export type ContextPreviewMode = 'auto' | 'manual' | 'strengthen' | 'preview';

export interface ContextBlockPreview {
  kind: string;
  title: string;
  enabled: boolean;
  reason: string | null;
  chars: number;
  estimated_tokens: number;
  source_count: number;
  max_chars: number | null;
  truncated: boolean;
  content_preview: string;
  meta: Record<string, unknown>;
}

export interface ContextPackPreview {
  meeting_id: number;
  mode: ContextPreviewMode;
  query_text: string;
  total_chars: number;
  estimated_tokens: number;
  max_chars: number | null;
  truncated: boolean;
  blocks: ContextBlockPreview[];
}

export async function getMeetingContextPreview(
  meetingId: number,
  params?: {
    mode?: ContextPreviewMode;
    q?: string;
    preview_chars_per_block?: number;
  },
): Promise<ContextPackPreview> {
  const query: Record<string, string | number> = {};
  if (params?.mode) query.mode = params.mode;
  if (params?.q) query.q = params.q;
  if (params?.preview_chars_per_block != null) query.preview_chars_per_block = params.preview_chars_per_block;
  const { data } = await api.get<ContextPackPreview>(`/meetings/${meetingId}/context-preview`, { params: query });
  return data;
}

File: meridian-web/backend/app/schemas/context_preview.py
"""Схемы предпросмотра Context Pack (Этап 6)."""

from typing import Any

from pydantic import BaseModel


class ContextBlockPreviewOut(BaseModel):
    kind: str
    title: str
    enabled: bool
    reason: str | None = None
    chars: int
    estimated_tokens: int
    source_count: int = 0
    max_chars: int | None = None
    truncated: bool = False
    content_preview: str
    meta: dict[str, Any] = {}


class ContextPackPreviewOut(BaseModel):
    meeting_id: int
    mode: str
    query_text: str
    total_chars: int
    estimated_tokens: int
    max_chars: int | None = None
    truncated: bool = False
    blocks: list[ContextBlockPreviewOut]
    # Этап 9.8: текущий авторитетный источник транскрипта (single|multi_channel) и эпохи
    transcription_source: str | None = None
    transcription_epochs_count: int = 0
    transcription_fallback_used: bool = False
